addons/mail/wizard/mail_compose_message.py

```diff
@@ -298,7 +298,7 @@ def get_mail_values(self, res_ids):
                 'parent_id': self.parent_id and self.parent_id.id,
                 'partner_ids': [partner.id for partner in self.partner_ids],
                 'attachment_ids': [attach.id for attach in self.attachment_ids],
-                'author_id': self.author_id.id,
+                'author_id': self.template_id.author_id.partner_id.id or self.author_id.id,
```

Comment:
I would keep self.author_id, and it would be overriden by values coming from generating author. Isn't is feasible ?